mev_inspect/arbitrages.py

```diff
@@ -78,7 +83,49 @@ def _get_arbitrages_from_swaps(swaps: List[Swap]) -> List[Arbitrage]:
         ]
 
 
-def _get_all_start_end_swaps(swaps: List[Swap]) -> List[Tuple[Swap, Swap]]:
+def _get_shortest_route(
+    start_swap: Swap,
+    end_swaps: List[Swap],
+    all_swaps: List[Swap],
+) -> Optional[List[Swap]]:
+    for end_swap in end_swaps:
+        if start_swap.token_out_address == end_swap.token_in_address:
+            return [start_swap, end_swap]
+
+    other_swaps = [
+        swap for swap in all_swaps if (swap is not start_swap and swap not in end_swaps)
+    ]
+
+    if len(other_swaps) == 0:
+        return None
+
+    shortest_route_rest = None
+
+    for next_swap in other_swaps:
+        if start_swap.token_out_address == next_swap.token_in_address and (
+            start_swap.contract_address == next_swap.from_address
+            or start_swap.to_address == next_swap.contract_address
+            or start_swap.to_address == next_swap.from_address
+        ):
+            shortest_from_next = _get_shortest_route(
+                next_swap,
+                end_swaps,
+                other_swaps,
+            )
+
+            if shortest_from_next is not None and (
+                shortest_route_rest is None
+                or len(shortest_from_next) < len(shortest_route_rest)
+            ):
+                shortest_route_rest = shortest_from_next
+
+    if shortest_route_rest is None:
+        return None
+    else:
+        return [start_swap] + shortest_route_rest
+
+
+def _get_all_start_end_swaps(swaps: List[Swap]) -> List[Tuple[Swap, List[Swap]]]:
```

Comment:
The return type indicates this will return a list of tuples containing a start swap and a list of corresponding plausible end swaps. The name/docstring seems a bit misleading that it would return 'pairs', as it's really returning all possible ends for each start swap.

Reply:
good call - will update

Reply:
(comments always out of sync with the codebase 😉)